### PR TITLE
test: Improved test coverage by adding tests of three classes

### PR DESCRIPTION
**Pull request by energinet-ajw**
Added tests to IntegrationEventTypeMapper, IntegrationEventService and CalculationResultCompletedToIntegrationEventFactory.

## References

* #887

### CODE REVIEW

File: source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs [hunk 1]
```diff
@@ -55,4 +55,34 @@ public class IntegrationEventDispatcherServiceTests
         // Assert
         integrationEventDispatcherMock.Verify(x => x.DispatchIntegrationEventsAsync(It.Is<int>(i => i > 0)));
     }
+
+    [Theory]
+    [AutoMoqData]
+    public async Task DispatchIntegrationEventsAsync_CallsDeleteOlderDispatchedIntegrationEventsOnIntegrationEventCleanUpService(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs:61]: Test name does not correlate with sut action.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs [hunk 2]
```diff
@@ -55,4 +55,34 @@ public class IntegrationEventDispatcherServiceTests
...
+        // Act
+        await sut.DeleteOlderDispatchedIntegrationEventsAsync(days);
+
+        // Assert
+        integrationEventCleanUpServiceMock.Verify(x => x.DeleteOlderDispatchedIntegrationEvents(days));
+    }
+
+    [Theory]
+    [AutoMoqData]
+    public async Task DeleteOlderDispatchedIntegrationEventsAsync_CallsCommitOnUnitOfWork(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs:77]: If you prefer a shorter name, here is a suggestion
```suggestion
    public async Task DeleteOlderDispatchedIntegrationEventsAsync_CommitsUnitOfWork(
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    [AutoMoqData]
+    public void CreateForEnergySupplier_ReturnAnIntegrationEventDtoWithTheCorrectValues(
+        [Frozen] Mock<ICalculationResultCompletedIntegrationEventFactory> calculationResultCompletedIntegrationEventFactoryMock,
+        [Frozen] Mock<IClock> clockMock,
+        [Frozen] Mock<IIntegrationEventTypeMapper> integrationEventTypeMapperMock,
+        CalculationResultCompleted calculationResultCompleted,
+        ProcessStepResult processStepResult,
+        ProcessCompletedEventDto processCompletedEventDto,
+        string energySupplierGln,
+        CalculationResultCompletedToIntegrationEventFactory sut)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:41]: How about this?
```suggestion
        IntegrationEventFactory sut)
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+        // Assert
+        Assert.Equal(calculationResultCompleted.GetType().ToString(), actual.MessageType);
+        Assert.Equal(calculationResultCompleted.ToByteArray(), actual.EventData);
+        Assert.Equal(instant, actual.CreationDate);
+    }
+
+    [Theory]
+    [AutoMoqData]
+    public void CreateForEnergySupplier_CallsCreateForBalanceResponsibleParty(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:67]:
```suggestion
    public void CreateForBalanceResponsibleParty_CallsCreateForBalanceResponsibleParty(
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+        // Act
+        sut.CreateForBalanceResponsibleParty(processStepResult, processCompletedEventDto, energySupplierGln);
+
+        // Assert
+        calculationResultCompletedIntegrationEventFactoryMock.Verify(x => x.CreateForBalanceResponsibleParty(processStepResult, processCompletedEventDto, energySupplierGln));
+    }
+
+    [Theory]
+    [AutoMoqData]
+    public void CreateForEnergySupplier_CallsCreateForGridArea(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:89]:
```suggestion
    public void CreateForTotalGridArea_CallsCreateForGridArea(
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+        // Act
+        sut.CreateForTotalGridArea(processStepResult, processCompletedEventDto);
+
+        // Assert
+        calculationResultCompletedIntegrationEventFactoryMock.Verify(x => x.CreateForGridArea(processStepResult, processCompletedEventDto));
+    }
+
+    [Theory]
+    [AutoMoqData]
+    public void CreateForEnergySupplier_CallsCreateForEnergySupplierByBalanceResponsibleParty(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:110]:
```suggestion
    public void CreateForEnergySupplierByBalanceResponsibleParty_CallsCreateForEnergySupplierByBalanceResponsibleParty(
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 7]
```diff
@@ -62,4 +62,30 @@ public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper su
         // Act & Assert
         Assert.Throws<KeyNotFoundException>(() => sut.GetMessageType(eventType));
     }
+
+    [Theory]
+    [AutoData]
+    public void ThrowsException_WhenAddingExistingEventType(IntegrationEventTypeMapper sut)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:68]:
```suggestion
    public void Add_WhenAddingExistingEventType_ThrowsException(IntegrationEventTypeMapper sut)
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 8]
```diff
@@ -62,4 +62,30 @@ public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper su
...
+    {
+        // Arrange
+        const string eventName = "eventName";
+        var eventType = typeof(CalculationResultCompleted);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:72]: You could have these autocreated. That would simplify the test and thus readability and maintainability.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 9]
```diff
@@ -62,4 +62,30 @@ public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper su
...
+        sut.Add(eventName, eventType);
+
+        // Act & Assert
+        Assert.Throws<ArgumentException>(() => sut.Add(eventName, eventType));
+    }
+
+    [Theory]
+    [AutoData]
+    public void ThrowsException_WhenAddingExistingEventName(IntegrationEventTypeMapper sut)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:81]:
```suggestion
    public void Add_WhenAddingExistingEventName_ThrowsException(IntegrationEventTypeMapper sut)
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+
+namespace Energinet.DataHub.Wholesale.WebApi.UnitTests.Infrastructure.EventPublishers;
+
+public class CalculationResultCompletedToIntegrationEventFactoryTests
+{
+    [Theory]
+    [AutoMoqData]
+    public void CreateForEnergySupplier_ReturnAnIntegrationEventDtoWithTheCorrectValues(
+        [Frozen] Mock<ICalculationResultCompletedIntegrationEventFactory> calculationResultCompletedIntegrationEventFactoryMock,
+        [Frozen] Mock<IClock> clockMock,
```
[the remark this answers is not in the document]
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:35]: What do you prefer? process completed event time or when event is created?

File: source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs (hunk 1 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs:61]: Changed.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 11]
```diff
@@ -62,4 +62,30 @@ public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper su
...
     }
+
+    [Theory]
+    [AutoData]
+    public void Add_WhenAddingExistingEventType_ThrowsException(IntegrationEventTypeMapper sut)
+    {
+        // Arrange
+        const string eventName = "eventName";
+        var eventType = typeof(CalculationResultCompleted);
+        sut.Add(eventName, eventType);
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:73]: Actually, I find it surprising to find an "Add" method on a mapper. What are your thoughts? Perhaps delegate to the ctor?

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:35]: It depends on how you see it. Does the timestamp reflect when the event instance was created or when the _event_ actually happened. Personally, I find the later most relevant. Although normally you are not possesion than a better option than to just select current timestamp. However, I'm suggesting it here for two reasons:
1. It'll simplify the class and the tests (one less dependency)
2. You actually have a timestamp at your disposal, which is closer to when the event actually happened

[the remark this answers is not in the document]
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:35]: I will keep as is :)

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs (hunk 8 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs]: Changed.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs [hunk 12]
```diff
@@ -55,4 +55,34 @@ public class IntegrationEventDispatcherServiceTests
         // Assert
         integrationEventDispatcherMock.Verify(x => x.DispatchIntegrationEventsAsync(It.Is<int>(i => i > 0)));
     }
+
+    [Theory]
+    [AutoMoqData]
+    public async Task DispatchIntegrationEventsAsync_CallsDeleteOlderDispatchedIntegrationEventsOnIntegrationEventCleanUpService(
+        [Frozen] Mock<IIntegrationEventCleanUpService> integrationEventCleanUpServiceMock,
+        IntegrationEventService sut)
```
[the remark this answers is not in the document]
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventDispatcherServiceTests.cs:63]: Changed.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 13]
```diff
@@ -62,4 +62,30 @@ public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper su
...
+    [AutoData]
+    public void ThrowsException_WhenAddingExistingEventName(IntegrationEventTypeMapper sut)
+    {
+        // Arrange
+        const string eventName = "eventName";
+        var eventType = typeof(CalculationResultCompleted);
+        sut.Add(eventName, eventType);
+
+        // Act & Assert
+        Assert.Throws<ArgumentException>(() => sut.Add(eventName, typeof(string)));
```
[the remark this answers is not in the document]
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs]: Changed.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs (hunk 11 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:74]: It has been implemented in the constructor but later moved out. Can't remember why. 
I have made add private and moved the add-logic to constructor.

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/CalculationResultCompletedToIntegrationEventFactoryTests.cs:35]: np. Just curious or learning: Why?

File: source/dotnet/wholesale-api/Infrastructure/EventPublishers/IntegrationEventTypeMapper.cs [hunk 14]
```diff
@@ -18,9 +18,13 @@ public class IntegrationEventTypeMapper : IIntegrationEventTypeMapper
...
 
-    public IntegrationEventTypeMapper()
+    public IntegrationEventTypeMapper(Dictionary<Type, string> dic)
     {
         _dic = new Dictionary<Type, string>();
+        foreach (var key in dic)
+        {
+            Add(key.Value, key.Key);
+        }
     }
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/Infrastructure/EventPublishers/IntegrationEventTypeMapper.cs:28]: How about simplifying things with this + removing the `Add` method?
```suggestion
    public IntegrationEventTypeMapper(Dictionary<Type, string> dic)
    {
        _dic = new Dictionary<Type, string>(dic);
    }
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventServiceTests.cs [hunk 15]
```diff
@@ -55,4 +55,34 @@ public class IntegrationEventDispatcherServiceTests
...
+    }
+
+    [Theory]
+    [AutoMoqData]
+    public async Task DeleteOlderDispatchedIntegrationEventsAsync_CommitsUnitOfWork(
+        [Frozen] Mock<IUnitOfWork> unitOfWorkMock,
+        int days,
+        IntegrationEventService sut)
+    {
+        // Arrange & Act
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventServiceTests.cs:82]:
```suggestion
        // Act
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventServiceTests.cs [hunk 16]
```diff
@@ -55,4 +55,34 @@ public class IntegrationEventDispatcherServiceTests
...
+
+    [Theory]
+    [AutoMoqData]
+    public async Task DeleteOlderDispatchedIntegrationEventsAsync_CallsDeleteOlderDispatchedIntegrationEventsOnIntegrationEventCleanUpService(
+        [Frozen] Mock<IIntegrationEventCleanUpService> integrationEventCleanUpServiceMock,
+        IntegrationEventService sut)
+    {
+        // Arrange
+        const int days = 10;
+
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Application/IntegrationEventsManagement/IntegrationEventServiceTests.cs:67]: nit: How about simplifying method body by making this a method input?

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 17]
```diff
@@ -40,26 +43,42 @@ public void GetEventName_WhenEventType_ReturnsEventName(IntegrationEventTypeMapp
 
     [Theory]
     [AutoData]
-    public void GetEventType_WhenEventName_ReturnsEventType(IntegrationEventTypeMapper sut)
+    public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper sut)
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:46]:
```suggestion
    public void GetMessageType_WhenGettingNotExisting_ThrowsException(IntegrationEventTypeMapper sut)
```

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 18]
```diff
@@ -40,26 +43,42 @@ public void GetEventName_WhenEventType_ReturnsEventName(IntegrationEventTypeMapp
...
-        // Assert
-        Assert.Equal(1, sut.Count());
+        // Act & Assert
+        Assert.Throws<KeyNotFoundException>(() => sut.GetMessageType(eventType));
     }
 
     [Theory]
     [AutoData]
-    public void ThrowsException_WhenGettingNotExisting(IntegrationEventTypeMapper sut)
+    public void OnInitialization_WhenAddingExistingEventType_ThrowsException(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:57]: I'm not sure but I think we normally name it like this (or perhaps it's just my personal preference). Feel free to decide what you prefer.
```suggestion
    public void Ctor_WhenAddingExistingEventType_ThrowsException(
```

Comment (BjarkeMeier) [reply source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:57]: image

File: source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs [hunk 19]
```diff
@@ -40,26 +43,42 @@ public void GetEventName_WhenEventType_ReturnsEventName(IntegrationEventTypeMapp
...
+        Assert.Throws<ArgumentException>(() => new IntegrationEventTypeMapper(new Dictionary<Type, string>
+        {
+            { type, eventName },
+            { type, otherEventName },
+        }));
+    }
 
+    [Theory]
+    [AutoData]
+    public void OnInitialization_WhenAddingExistingEventName_ThrowsException(
```
Comment (BjarkeMeier) [inline source/dotnet/wholesale-api/WebApi.UnitTests/Infrastructure/EventPublishers/IntegrationEventTypeMapperTests.cs:72]:
```suggestion
    public void Ctor_WhenAddingExistingEventName_ThrowsException(
```

File: source/dotnet/wholesale-api/Infrastructure/EventPublishers/IntegrationEventTypeMapper.cs (hunk 14 above)
Comment (energinet-ajw) [reply source/dotnet/wholesale-api/Infrastructure/EventPublishers/IntegrationEventTypeMapper.cs:28]: If I remove the Add the validation is fired.

**Review by BjarkeMeier — APPROVED** [review]

**Pull request merged**